sapling-crypto/src/bundle.rs

```diff
@@ -331,19 +319,8 @@ impl<A: Authorization> std::fmt::Debug for SpendDescription<A> {
 }
 
 impl<A: Authorization> SpendDescription<A> {
-    #[cfg(feature = "temporary-zcashd")]
-    pub fn temporary_zcashd_from_parts(
-        cv: ValueCommitment,
-        anchor: bls12_381::Scalar,
-        nullifier: Nullifier,
-        rk: redjubjub::VerificationKey<SpendAuth>,
-        zkproof: A::SpendProof,
-        spend_auth_sig: A::AuthSig,
-    ) -> Self {
-        Self::from_parts(cv, anchor, nullifier, rk, zkproof, spend_auth_sig)
-    }
-
-    pub(crate) fn from_parts(
+    /// Constructs a v4 `SpendDescription` from its constituent parts.
+    pub fn from_parts(
```

Comment:
Now that these functions are public, rather than just pub(crate), are there invariants that should be enforced in them? It's fine to do this as a follow-up in the `sapling-crypto` crate, but if such work should be done then please open an issue for it.

Reply:
I added the necessary invariant to `Bundle::from_parts`, but I couldn't at the time think of any other invariants that were necessary for `SpendDescription` and `OutputDescription`. I'll think it over before we cut a release of `sapling-crypto`.